### PR TITLE
fix(#26): add warning if jsonDir contains file when running csv-to-doc command

### PR DESCRIPTION
**Pull request by sugat009**
# Description

Added warning if `jsonDir` contains files when running `csv-to-doc` command.

medic/cht-conf#26

# Screenshots
1. When `jsonDir` does not exists
image
2. When `jsonDir` exists and user wants to continue after prompt is shown
image
3. When `jsonDir` exists and user does not want to continue after prompt is shown
image

# Code review items

- Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- Tested: Unit and/or integration tests where appropriate
- Backwards compatible: Works with existing data and configuration. Any breaking changes documented in the release notes.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: src/lib/sync-fs.js
Comment (jkuester) [inline src/lib/sync-fs.js:84]:
```suggestion
  return !fs.readdirSync(dir).length;
```

Minor, but no need to use a ternary to get a boolean value.

File: test/lib/sync-fs.spec.js
Comment (jkuester) [inline test/lib/sync-fs.spec.js:10]: Let's just combine this with the `require`:

```js
const { expect } = require('chai');
```

Also, we try not to mix `expect` and `assert` in the same files.  The go-forward pattern is to use `expect`, so can you just remove `assert` and update that test to use `expect`? (Just in this file...)

File: src/fn/csv-to-docs.js
Comment (jkuester) [inline src/fn/csv-to-docs.js:39]: This `execute` function is already criminally large.  Can you extract this block into a new little function called something like `warnIfDirectoryHasDocs` and just call the function here?

**Review by jkuester — CHANGES_REQUESTED** [review]
So great!  You did an awesome job of paying attention to the patterns and functionality of existing code and matching that with your new additions!

Can you also add two tests to the `csv-to-docs.spec.js` file, one where the user accepts the prompt and one where the user rejects it?  I think you should be able to mock the `userPrompt` functionality with sinon similar to what is happening in the [`edit-contacts.spec` tests](https://github.com/medic/cht-conf/blob/41423656a7393a4bb6c4f207c3624639b0f1a760/test/fn/edit-contacts.spec.js#L299).

**Comment by jkuester** [comment]
Also, I added @tatilepizs to this PR so we can get QA's perspective here too!

**Comment by andrablaj** [comment]
@jkuester, my bad for not having @tatilepizs added to this PR, and thanks for doing that. @sugat009 asked me about it this morning, and as it was a small change and the PR was already ready for review I figured that it didn't need extra QA support.

**Review by tatilepizs — APPROVED** [review]
I don't have anything else to add.
Thank you @sugat009 ⭐

File: test/lib/sync-fs.spec.js
Comment (sugat009) [reply test/lib/sync-fs.spec.js:8]: Updated.

File: src/fn/csv-to-docs.js
Comment (sugat009) [reply src/fn/csv-to-docs.js]: updated.

File: test/lib/sync-fs.spec.js
Comment (m5r) [inline test/lib/sync-fs.spec.js:5]: Could we clear up a bit the confusion between these two imports? Maybe rename the first one to `syncFs` or rename the second one to `nodeFs`

I noticed that `sync-fs.js` re-exports node's `fs` module so we could also replace all `fs_` occurrences with `fs.fs` and drop the `const fs_ require('fs');` line

Either solution is perfectly fine, the goal is just to clear up any potential confusion with variable names

**Review by m5r — DISMISSED** [review]
Well done on landing your first PR for cht-conf, @jkuester and @tatilepizs covered pretty much everything I had in mind :smile: 
I added a few comments, let me know if you have any question

File: test/lib/sync-fs.spec.js
Comment (sugat009) [reply test/lib/sync-fs.spec.js:5]: Thanks for pointing this out. 
The pattern I've seen in the code is that when `sync-fs` is imported it's assigned to a variable named as `fs` as it is done here as well. However, for this test suite I need to mock the nod JS's `fs` which is also assigned to a variable named as `fs` but in this file `fs` name had already been taken so I assigned it to `fs_`. If I change the first `fs` variable then, the PR would have lots of changes unrelated to the issue in question and I wanted to keep it minimal and I named it as `fs_` because it would not in anyway mess up with the autocomplete of the IDE.

> I noticed that sync-fs.js re-exports node's fs module so we could also replace all fs_ occurrences with fs.fs and drop the const fs_ require('fs'); line

I am more aligned with this suggestion as it does not require changes for the previous code.
What do you think @m5r , @jkuester ?

Comment (jkuester) [reply test/lib/sync-fs.spec.js:5]: @sugat009 I think your rational makes sense!  Leave the first `fs` alone. As long as sinon works fine with it, just change `fs_` to `fs.fs`.  If sinon gives you trouble, just change `fs_` to `nodeFs`.  

(Also, Mokhtar is OOO next week.  I don't think we need to wait this PR on his feedback. So, I think we should proceed with your plan!)

Comment (sugat009) [reply test/lib/sync-fs.spec.js]: thanks @jkuester , updated.

File: test/lib/sync-fs.spec.js
Comment (jkuester) [reply test/lib/sync-fs.spec.js:5]: :point_up: Looks like maybe this did not get updated yet!

File: src/fn/csv-to-docs.js
Comment (jkuester) [inline src/fn/csv-to-docs.js:281]: In my opinion, it is a bad pattern to export a function just so that it can be unit tested (though admittedly we do it quite a bit in this codebase.  I just hate to add to it....). It makes it more confusing for the reader to understand what functionality the file is actually offering.  A better approach would be:

- Just test this function as part of the larger workflow by calling `execute`.
- Or, extract this function to a lib file where it could be reused by other files in the future (this functionality seem generic enough that it could be reused for other workflows/actions).  The `lib/sync-fs.js` file seems like a reasonable place to put it. Then, the tests in the `csv-to-docs.spec.js` can mock this functionality and test both cases without caring about the internals of the function...

**Review by jkuester — CHANGES_REQUESTED** [review]
Sorry, I think when I first looked at this, I was looking at an older version of the code and I did not see that `warnIfDirectoryIsNotEmpty` was getting exported.  (Which probably made at least some of my comments more confusing...)

File: src/fn/csv-to-docs.js
Comment (sugat009) [reply src/fn/csv-to-docs.js:281]: Yes, I referenced some of the older code before doing this.
I'd go with the 2nd option, I was considering it before doing what I did, just wasn't sure where I should be keeping it.

Comment (sugat009) [reply src/fn/csv-to-docs.js:281]: I've gone with the 2nd option @jkuester suggested.
However, I've encountered a small `gotcha` . When moving the `warnIfDirectoryIsNotEmpty` function to `sync-fs.js` file, both `warnIfDirectoryIsNotEmpty` and `isDirectoryEmpty`(function used by `warnIfDirectoryIsNotEmpty`) will be located within the same module, so it won't be possible to mock `isDirectoryEmpty` anymore(Reference: https://github.com/sinonjs/sinon/issues/1161#issuecomment-251591197). Hence, when testing `warnIfDirectoryIsNotEmpty`, I will not be mocking `isDirectoryEmpty` anymore and will start mocking `fs.readdirSync`(implementation inside `isDirectoryEmpty`). So, the tests won't exactly be unit tests but kind of a integration test.
Let me know if it's acceptable.

Comment (jkuester) [reply src/fn/csv-to-docs.js:281]: :+1: This seems fine to me. I you look for perfection in JS you will always be disappointed!  :sweat_smile: Your proposal seems like the best balance of good structure and good testing!

Comment (sugat009) [reply src/fn/csv-to-docs.js]: Thanks, @jkuester . Updated.

File: test/lib/sync-fs.spec.js
Comment (sugat009) [reply test/lib/sync-fs.spec.js]: Updated, now. :)

File: test/fn/csv-to-docs.spec.js
Comment (jkuester) [inline test/fn/csv-to-docs.spec.js:38]: Okay, we have great tests for the internals of the `sync-fs.warnIfDirectoryIsNotEmpty` function. The last piece to have total test coverage is just to validate that the function is actually called during the `execute`.  Should be able to stub the function and just assert here that it was called with the expected warning message.

**Review by jkuester — APPROVED** [review]
This is great! Thanks for the careful revisions!  I just had one more minor request for an assertion in the `csv-to-docs.spec.js`.  I am going to approve this PR now, though, meaning that I don't think I need to review how you implemented the changes I requested (this is a common practice at Medic, especially for minor test changes like this. Just helps reduce unnecessary feedback loops since we are all working async).  If you do have any questions/concerns with the changes, you can always re-request my review.  Otherwise, if everything looks good to you, once you get that assertion added, this PR is good to merge (already have approval from QA)! 

You should have permissions to click the merge button.  Just make sure it is doing a "squash merge" and the commit message is following [the convention](https://github.com/medic/cht-conf?tab=readme-ov-file#commit-format). Both of these should be done automatically based on our repo settings (which is why we always merge to `main` via the PR webpage on GitHub (and not via the CLI or anything) since it will enforce this stuff).  The commit message format is very important since merging to master will trigger the `semantic-release` plugin to try and automatically cut a new release of cht-conf based on the info in the commit message.

**Pull request merged**
